account_financial_risk/tests/test_account_financial_risk.py

```diff
@@ -210,20 +210,16 @@ def test_recompute_newid(self):
         new._compute_risk_account_amount()
 
     def test_batch_invoice_confirm(self):
-        self.invoice._post()
+        self.invoice.action_post()
         line = self.invoice.line_ids.filtered(lambda x: x.debit != 0.0)
         line.date_maturity = "2017-01-01"
         self.partner.risk_invoice_unpaid_include = True
         self.partner.credit_limit = 100.0
         invoice2 = self.invoice.copy({"partner_id": self.invoice_address.id})
-        validate_wiz = (
-            self.env["validate.account.move"]
-            .with_context(active_model="account.move", active_ids=invoice2.ids)
-            .create({})
-        )
         with self.assertRaises(ValidationError):
-            invoice2._post()
-            validate_wiz.validate_move()
+            invoice2.with_context(
```

Comment:
This is not doing the same, as we want to try the batch move validation.

Reply:
Uhm ok... there is another way you know for doing the same? As this wizard works with `_post` method

Reply:
@carlosdauden you are the one that know this more. Can you give your advice?

Reply:
Yes, keep in mind that the restriction must work both from the form view button and when multiple invoices are selected from the list view. Odoo doesn't offer the best implementations to extend methods...
I'll take a look at it next week.

Reply:
the only way is to inherit both `action_post` and `_post`... I'll try and back to you asap

Reply:
@pedrobaeza @carlosdauden I changed implementation to cover invoice confirmation from both form and list.